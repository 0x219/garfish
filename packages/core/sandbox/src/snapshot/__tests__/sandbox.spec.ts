import { SnapshotSandbox } from '../index';

describe('test sandbox ', () => {
  it('dom sandbox', () => {
    document.head.innerHTML = `
      <style>body{color: 'yellow';}</style>
    `;

    const obj = {
      name: 'bytedance',
      age: '8',
      some: 'test delete',
    };

    const sb = new SnapshotSandbox('app1', obj);

    const st = document.createElement('style');
    st.style.cssText = 'background: red;';
    document.head.appendChild(st);
    obj.name = 'wu';
    obj.age = '23';
    delete obj.some;

    sb.deactivate();

    const result = document.querySelectorAll('style');
    expect(result).toMatchSnapshot();
    expect(obj).toMatchSnapshot();
  });
});
