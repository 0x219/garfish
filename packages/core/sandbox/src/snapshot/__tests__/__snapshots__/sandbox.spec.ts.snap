// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test sandbox  dom sandbox 1`] = `
NodeList [
  <style>
    body{color: 'yellow';}
  </style>,
]
`;

exports[`test sandbox  dom sandbox 2`] = `
Object {
  "age": "8",
  "name": "bytedance",
  "some": "test delete",
}
`;
